frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/ResetCurrentToParentAction.java

```diff
@@ -0,0 +1,86 @@
+package com.virtuslab.gitmachete.frontend.actions.traverse;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+import static org.checkerframework.checker.i18nformatter.qual.I18nConversionCategory.GENERAL;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.collection.List;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.i18nformatter.qual.I18nFormat;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.tainting.qual.Untainted;
+
+import com.virtuslab.gitmachete.backend.api.IBranchReference;
+import com.virtuslab.gitmachete.backend.api.ILocalBranchReference;
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.base.BaseResetAction;
+import com.virtuslab.gitmachete.frontend.actions.base.IBranchNameProvider;
+import com.virtuslab.gitmachete.frontend.actions.base.ISyncToParentStatusDependentAction;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+public class ResetCurrentToParentAction extends BaseResetAction
```

Comment:
Hmmm _Reset to Parent_... in what case is that action suggested to user? it seems a rather unusual 🤔

Reply:
one of the options when branch is out of sync to parent.

Reply:
Yeah, discussed there, I strongly discourage ☹️